libs/payments/ui/src/lib/client/components/CheckoutForm.tsx

```diff
@@ -135,71 +146,101 @@ export function CheckoutForm({ readOnly, cart, locale }: CheckoutFormProps) {
+        }
+        onClick={() => setShowConsentError(true)}
+      >
+        <Localized id="next-new-user-card-title">
+          <h3 className="font-semibold text-grey-600 text-start">
+            Enter your card information
+          </h3>
+        </Localized>
+        {!isPaymentElementLoading && (
+          <Form.Field
+            name="name"
+            serverInvalid={hasFullNameError}
+            className="my-6"
+          >
+            <Form.Label className="text-grey-400 block mb-1 text-start">
+              <Localized id="payment-name-label">
+                Name as it appears on your card
+              </Localized>
+            </Form.Label>
+            <Form.Control asChild>
+              <input
+                className="w-full border rounded-md border-black/30 p-3 placeholder:text-grey-500 placeholder:font-normal focus:border focus:!border-black/30 focus:!shadow-[0_0_0_3px_rgba(10,132,255,0.3)] focus-visible:outline-none data-[invalid=true]:border-alert-red data-[invalid=true]:text-alert-red data-[invalid=true]:shadow-inputError"
+                type="text"
+                data-testid="name"
+                placeholder={l10n.getString(
+                  'payment-name-placeholder',
+                  {},
+                  'Full Name'
+                )}
+                readOnly={!formEnabled}
+                tabIndex={formEnabled ? 0 : -1}
+                value={fullName}
+                onChange={(e) => {
+                  setFullName(e.target.value);
+                  setHasFullNameError(!e.target.value);
+                }}
+                aria-required
+              />
+            </Form.Control>
+            {hasFullNameError && (
+              <Form.Message asChild>
+                <Localized id="next-payment-validate-name-error">
+                  <p className="mt-1 text-alert-red" role="alert">
+                    Please enter your name
+                  </p>
+                </Localized>
+              </Form.Message>
+            )}
+          </Form.Field>
+        )}
+        <PaymentElement
+          options={{
+            layout: {
+              type: 'accordion',
+              defaultCollapsed: false,
+              radios: false,
+              spacedAccordionItems: true,
+            },
+            readOnly: !formEnabled,
+          }}
+        />
+        {!isPaymentElementLoading && (
+          <Form.Submit asChild>
             <PrimaryButton
               type="submit"
               aria-disabled={
                 !stripeFieldsComplete || !nonStripeFieldsComplete || loading
               }
+              className={
+                formEnabled
+                  ? ''
+                  : 'cursor-not-allowed relative focus:border-blue-400 focus:outline-none focus:shadow-input-blue-focus after:absolute after:content-[""] after:top-0 after:left-0 after:w-full after:h-full after:bg-white after:opacity-50 after:z-10'
+              }
```

Comment:
This shouldn't be necessary. The <div> creating the disabled layer encapsulates the button, so the `cursor-not-allowed` should also apply to the button.

I had a look at SP2, and it looks like the `PrimaryButton` is missing class z-[5], which we can probably replace with z-10. Adding that should work. (Note, this can be added to the `PrimaryButton` component directly)